platformDataBackendDisease.sc

```diff
@@ -0,0 +1,172 @@
+import $ivy.`com.typesafe:config:1.3.4`
+import $ivy.`com.github.fommil.netlib:all:1.1.2`
+import $ivy.`org.apache.spark::spark-core:2.4.3`
+import $ivy.`org.apache.spark::spark-mllib:2.4.3`
+import $ivy.`org.apache.spark::spark-sql:2.4.3`
+import $ivy.`com.github.pathikrit::better-files:3.8.0`
+import $ivy.`sh.almond::ammonite-spark:0.7.0`
+import org.apache.spark.SparkConf
+import org.apache.spark.sql._
+import org.apache.spark.sql.types._
+import org.apache.spark.sql.functions._
+import org.apache.spark.sql.functions.col
+
+
+object Loaders {
+  /** Load Diseases file */
+  def loadDiseases(path: String)(implicit ss: SparkSession): DataFrame = {
+    val diseaseList = ss.read.json(path)
+    diseaseList
+  }
+
+  /** Load Evidences file */
+  def loadEvidences(path: String)(implicit ss: SparkSession): DataFrame = {
+    val evidences = ss.read.json(path)
+    evidences
+  }
+}
+
+object Transformers {
+  implicit class Implicits (val df: DataFrame) {
+
+    def setIdAndSelectFromDiseases(ss: SparkSession): DataFrame = {
+      val genAncestors = udf((codes: Seq[Seq[String]]) =>
+        codes.view.flatten.toSet.toSeq)
+
+      val efos_summary = df
+        .withColumn("id", substring_index(col("code"), "/", -1))
+        .withColumn("ancestors", genAncestors(col("path_codes")))
+        .withColumn(
+          "isTherapeuticArea", size(genAncestors(col("path_codes"))) === 1).as("isTherapeuticArea")
```

Comment:
Good, just to rewrite as I said before and you dont need to use an alias as the function `withColumn` already gives a name to the computed column
```scala
.withColumn(
          "isTherapeuticArea", size(flatten(col("path_codes"))) === 1)
```